test/e2e/kubevirt.go

```diff
@@ -0,0 +1,628 @@
+				Should(Succeed(), step)
+
+			checkConnectivity(td, vm.Name, endpoints, "before live migration")
+
+			for i := uint(1); i <= td.numberOfMigrations; i++ {
+
+				by(vm.Name, fmt.Sprintf("Live migrate virtual machine #%d", i))
+				liveMigrateVirtualMachine(vm.Name, td.mode)
+
+				checkConnectivity(td, vm.Name, endpoints, fmt.Sprintf("after live migration #%d", i))
+			}
+
+		}
+	)
+	DescribeTable("when live migrated", func(td liveMigrationTestData) {
+		if td.mode == kubevirtv1.MigrationPostCopy && os.Getenv("GITHUB_ACTIONS") == "true" {
+			Skip("Post copy live migration not working at github actions")
+		}
+		var (
+			err error
+		)
+
+		namespace = fr.Namespace.Name
+
+		kvcli, err = newKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+
+		bandwidthPerMigration := resource.MustParse("40Mi")
+		forcePostCopyMigrationPolicy := &kvmigrationsv1alpha1.MigrationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "force-post-copy",
+			},
+			Spec: kvmigrationsv1alpha1.MigrationPolicySpec{
+				AllowPostCopy:           pointer.Bool(true),
+				CompletionTimeoutPerGiB: pointer.Int64(1),
+				BandwidthPerMigration:   &bandwidthPerMigration,
+				Selectors: &kvmigrationsv1alpha1.Selectors{
+					VirtualMachineInstanceSelector: kvmigrationsv1alpha1.LabelSelector{
+						"test-live-migration": "post-copy",
+					},
+				},
+			},
+		}
+		if td.mode == kubevirtv1.MigrationPostCopy {
+			_, err = kvcli.MigrationPolicy().Create(context.TODO(), forcePostCopyMigrationPolicy, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			defer func() {
+				Expect(kvcli.MigrationPolicy().Delete(context.TODO(), forcePostCopyMigrationPolicy.Name, metav1.DeleteOptions{})).To(Succeed())
+			}()
+		}
+
+		By("Creating a test pod on both selected worker nodes")
+		podWorker1 = composeAgnhostPod(
+			"testpod-ovn-worker",
+			namespace,
+			"ovn-worker",
+			"/bin/bash", "-c", "/agnhost netexec --http-port 8000")
+		podWorker2 = composeAgnhostPod(
+			"testpod-ovn-worker2",
+			namespace,
+			"ovn-worker2",
+			"/bin/bash", "-c", "/agnhost netexec --http-port 8000")
+		_ = fr.PodClient().CreateSync(podWorker1)
+		_ = fr.PodClient().CreateSync(podWorker2)
+
+		By("Waiting until both pods have an IP address")
+		Eventually(func() error {
+			var err error
+			podWorker1, err = fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Get(context.TODO(), podWorker1.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if podWorker1.Status.PodIP == "" {
+				return fmt.Errorf("pod %s has no valid IP address yet", podWorker1.Name)
+			}
+			podWorker2, err = fr.ClientSet.CoreV1().Pods(fr.Namespace.Name).Get(context.TODO(), podWorker2.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if podWorker2.Status.PodIP == "" {
+				return fmt.Errorf("pod %s has no valid IP address yet", podWorker2.Name)
+			}
+			return nil
+		}).
+			WithTimeout(time.Minute).
+			WithPolling(time.Second).
+			Should(Succeed())
```

Comment:
could you iterate on something and avoid repeating this code. 

At least the "creation" part of it; then you'd block until both pod workers are ready.